x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable_factory.ts

```diff
@@ -105,4 +106,15 @@ export class EmbeddableFactory implements EmbeddableFactoryDefinition {
       parent
     );
   }
+
+  extract(state: EmbeddableStateWithType) {
+    let references: SavedObjectReference[] = [];
+    const typedState = (state as unknown) as LensEmbeddableInput;
+
+    if ((typedState as LensByValueInput).attributes !== undefined) {
```

Comment:
nit: You can get around these two type casts by checking for by value input like this:
`'attributes' in typedState && typedState.attributes`

It's turned into a type guard automatically by typescript, so you won't need to case again in the branch

Reply:
Ohh nice. That's very handy to know.